cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternals.qll

```diff
@@ -192,20 +198,17 @@ private predicate sourceVariableHasBaseAndIndex(SourceVariable v, BaseSourceVari
   v.getIndirection() = ind
 }
 
-class DefImpl extends DefOrUseImpl, TDefImpl {
+abstract class DefImpl extends DefOrUseImpl {
   Operand address;
   int ind;
 
-  DefImpl() { this = TDefImpl(address, ind) }
+  bindingset[ind]
+  DefImpl() { any() }
```

Comment:
Is this needed, since the class is now abstract.

Reply:
Yep:
```ql
abstract class E extends Expr {
  int n;
}
```
complains about `n` not being bound. It's kinda strange that the compiler is requiring this here since it's already requiring it for all the extents of the class. I think this has been raised to the Foundations team before, but I can't remember if they had a good answer 😂 (they probably did, though!)